data/scripts/creaturescripts/modal_window_helper.lua

```diff
@@ -0,0 +1,197 @@
+local ModalWindowAutoID = 10000
+local buttonOrder = {
+    [4] = {2, 3, 4, 1},
+    [3] = {2, 3, 1},
+    [2] = {2, 1},
+    [1] = {1}
+}
+
+if not __ModalWindow then
+    __ModalWindow = ModalWindow
+    ModalWindows = {}
+end
+
+ModalWindow = {}
+ModalWindow.__index = ModalWindow
+
+function ModalWindow.new(self, ...)
+    if type(self) ~= "table" then
+        return __ModalWindow(self, ...)
+    end
+
+    self.modalWindowId = 0
+    self.buttons = {}
+    self.choices = {}
+    self.using = 0
+    return setmetatable(self, ModalWindow)
+end
+
+setmetatable(ModalWindow, {
+    __call = function (self, ...)
+        return ModalWindow.new(...)
+    end
+})
+
+function ModalWindow:setTitle(title)
+    self.title = title
+end
+
+function ModalWindow:setMessage(message)
+    self.message = message
+end
+
+function ModalWindow:addButton(name, callback)
+    for _, button in pairs(self.buttons) do
+        if button.name == name then
+            button.callback = callback
+            return
+        end
+    end
+    self.buttons[#self.buttons + 1] = {name = name, callback = callback}
+end
+
+function ModalWindow:removeButton(name)
+    for i = 1, #self.buttons do
+        if self.buttons[i].name == name then
+            table.remove(self.buttons, i)
+            break
+        end
+    end
+end
+
+function ModalWindow:callButton(name, player, button, choice)
+    for _, button in pairs(self.buttons) do
+        if button.name == name then
+            button.callback(player, button, choice)
+            return
+        end
+    end
+end
+
+function ModalWindow:clearButtons()
+    self.buttons = {}
+end
+
+function ModalWindow:setDefaultEnterButton(buttonId)
+    self.defaultEnterButton = buttonId
```

Comment:
Can't we just use `modalWindow.defaultEnterButton = buttonId` instead of calling a function?

Reply:
It can be done, so of course. I can remove the function.

Reply:
Rename the function argument to `buttonName` and also add the return to each function so we can use them to validate `IF`

I thought about removing it, but what if someone wants to use the classic methods of the ModalWindow meta table but at the same time wants to use this helper system? It would be strange that if you use ModalWindow by default there is a method called `setDefaultEnterButton` but when you use the helper system you get an error for trying to access a field with a null value and trying to call it.